manager/src/main/java/com/redhat/service/bridge/manager/dao/ConnectorsDAO.java

```diff
@@ -20,9 +20,13 @@ public ConnectorEntity findByProcessorIdAndName(String processorId, String name)
         return singleResultFromList(find("#CONNECTORENTITY.findByProcessorIdAndName", p));
     }
 
-    public ConnectorEntity findByProcessorId(String processorId) {
+    public List<ConnectorEntity> findUnprocessed() {
+        return find("#CONNECTORENTITY.findUnprocessed").list();
+    }
+
+    public List<ConnectorEntity> findByProcessorId(String processorId) {
```

Comment:
Is this going to return always a list with just one item?

Reply:
It's a `OneToOne` mapped with a `OneToMany/ManyToOne` so theoretically yes, even though in practice it's impossible. The reason for which I haven't used `OneToOne` is that in Hibernate they're usually complicated and it should be avoided (please excuse me why I don't write the exact reasons , I need to check again but it's something I've memorized) :)

I'd rather not assume in the code that we might not have by error two connectors per processor, so I think it's better to map it with a list anyway.